### PR TITLE
connectInfiniteHits: Fix fetch the first page

### PR DESCRIPTION
**Pull request by Kikobeats**
Because first page index is 0

### CODE REVIEW

**Comment by vvo** [comment]
The state that is here is the state we "Expose". When doing queries to algolia we then do page - 1 because algolia is 0 based.

But inside the state, state.page = 1 means we will query page 0 on Algolia.

Could you walk us through your usecase/struggle with this maybe? How did you encounter a bug, how can we reproduce it? Thanks

**Comment by Kikobeats** [comment]
I found this behavior playing in [windtoday.co](https://windtoday.co/) with facets. 

Currently is monkey patched on production, but basically in the results:

- I see the page 1.
- When I click in `load more` it injects the results of the page 1 again.

**Comment by Kikobeats** [comment]
@vvo I revert my production build to make more easy reproduce the issue.

Enter in [this url](https://windtoday.co/?range%5Bsail.size%5D%5Bmin%5D=6&range%5Bsail.size%5D%5Bmax%5D=7&p=0).

Now check the title of the results and click in `Load More`. You can find duplicate titles.

File: packages/react-instantsearch/src/connectors/connectInfiniteHits.js [hunk 1]
```diff
@@ -66,7 +66,7 @@ export default createConnector({
     const id = getId();
     const nextPage = searchState[id] ?
       Number(searchState[id]) + 1 :
-      1;
+      0;
```
Comment (bobylito) [inline packages/react-instantsearch/src/connectors/connectInfiniteHits.js]: Looks legit based on Algolia API knowledge, but we've had discussion and we decided to fix the Algolia API and provide a user understandable numeration. If there is a bug, it lies where the correction is done not here.

Comment (Kikobeats) [inline packages/react-instantsearch/src/connectors/connectInfiniteHits.js]: I'm agree, this is a monkey patch, but looks a problem related with what index is considered as first index

**Comment by Kikobeats** [comment]
Hmm I need to investigate more because after this change I still see duplicate fetch pages 🤔

**Comment by Kikobeats** [comment]
I updated the PR, for me have more sense and resolve the problem.

What do you think, @vvo @bobylito ?

**Comment by Kikobeats** [comment]
nope, looks weird. xD

**Comment by Kikobeats** [comment]
Currently just resolved adding

```
searchParameters={{hitsPerPage: 15}}
```

without modify the code.

**Comment by vvo** [comment]
@Kikobeats Do you think you could either:
- write a unit test
- create a simple example

Using this repository? So that we reduce the test case to the only parts that are relevant?

**Comment by vvo** [comment]
@Kikobeats I fail at seeing if we have a real bug here or what? Having a simple example could help, otherwise we'll close this untill someone find a simple example to reproduce it

**Comment by Kikobeats** [comment]
Sorry currently I fixed following the `searchParameter` setup, I need to investigate more for open a new issue

**Comment by bobylito** [comment]
Thanks @Kikobeats for taking the time!

**Pull request closed without merging**